### PR TITLE
pythonPackages.pyhocon: init at 0.3.51

### PR DESCRIPTION
**Pull request by chreekat**
###### Motivation for this change

Add pyhocon package.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by FRidh** [comment]
@GrahamcOfBorg build python3.pkgs.pyhocon python2.pkgs.pyhocon

**Comment by chreekat** [comment]
My [patch to make tests possible](https://github.com/chimpler/pyhocon/pull/203) has been merged upstream, but not released to PyPi yet.

File: pkgs/development/python-modules/pyhocon/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  version = "0.3.51";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "10l014br012fa583rnj3wqf6g9gmljamcwpw4snqwwg15i0dmkll";
+  };
+
+  propagatedBuildInputs = [ pyparsing ];
+
+  checkInputs = [ setuptools pytest mock ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/pyhocon/default.nix]:
```suggestion
  checkInputs = [ pytest mock ];
```

File: pkgs/development/python-modules/pyhocon/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  # Tests fail because necessary data files aren't packaged for PyPi yet.
+  # See https://github.com/chimpler/pyhocon/pull/203
+  doCheck = false;
+
+  meta = with lib; {
+    homepage = https://github.com/chimpler/pyhocon/;
+    description = "HOCON parser for Python";
+    # Long description copied from <GitHub>/setup.py on 2019-03-06.
```
Comment (dotlambda) [inline pkgs/development/python-modules/pyhocon/default.nix]:
```suggestion
```

File: pkgs/development/python-modules/pyhocon/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+, fetchPypi
+# Runtime inputs:
+, pyparsing
+# Check inputs:
+, setuptools
+, pytest
+, mock
+}:
+
+buildPythonPackage rec{
```
Comment (dotlambda) [inline pkgs/development/python-modules/pyhocon/default.nix]:
```suggestion
buildPythonPackage rec {
```

File: pkgs/development/python-modules/pyhocon/default.nix [hunk 4]
```diff
@@ -0,0 +1,41 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+# Runtime inputs:
+, pyparsing
+# Check inputs:
+, setuptools
```
Comment (dotlambda) [inline pkgs/development/python-modules/pyhocon/default.nix]:
```suggestion
```

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/pyhocon/default.nix (hunk 2 above)
Comment (chreekat) [reply pkgs/development/python-modules/pyhocon/default.nix]: I'd rather keep this. Only slightly worse than no long description at all is a copy/pasted description with no reference cited. :) However, I don't like my shorthand, so I'll be more explicit.

**Comment by chreekat** [comment]
Thanks for the review :+1:

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.pyhocon python3.pkgs.pyhocon

**Pull request merged**